docs: [OpenAI] Update documentation and release notes for new api (#354)
SAP/ai-sdk-java: docs/guides/OPENAI_CHAT_COMPLETION.md

## Message history

**Since v1.4.0**

```java

> var request =
>     new OpenAiChatCompletionRequest(
>         OpenAiMessage.system("I'm fine, thank you!"),
>         OpenAiMessage.user("Hello, how are you?"));

Reviewer: Maybe we should use more real-world like examples here? This makes it seem like the system message is in response to the user message.
```suggestion
var request =
    new OpenAiChatCompletionRequest(
        OpenAiMessage.system("You are a helpful assistant"),
        OpenAiMessage.user("Hello World! Why is this phrase so famous?"));
```

}
```

### Asynchronous Streaming - Non-blocking

**Since v1.4.0**

> The following example demonstrate how you can leverage a concurrency-safe container (like an AtomicReference) to "listen" for usage information in
> any incoming delta.

Reviewer: Weird line-break

```suggestion
The following example demonstrate how you can leverage a concurrency-safe container (like an AtomicReference) to "listen" for usage information in
any incoming delta.
```

```java
String question = "Can you give me the first 100 numbers of the Fibonacci sequence?";
var userMessage = OpenAiMessage.user(question);
var request = new OpenAiChatCompletionRequest(userMessage);

OpenAiClient client = OpenAiClient.forModel(GPT_4O);
var usageRef = new AtomicReference<CompletionUsage>();

// Prepare the stream before starting the thread to handle any initialization exceptions
Stream<OpenAiChatCompletionDelta> stream = client.streamChatCompletionDeltas(request);

// Create a new thread for asynchronous, non-blocking processing
Thread streamProcessor =
    new Thread(
        () -> {
            // try-with-resources ensures the stream is closed after processing
            try (stream) {
                stream.forEach(
                    delta -> {
                        usageRef.compareAndExchange(null, delta.getCompletionUsage());
                        System.out.println("Content: " + delta.getDeltaContent());
                    });
            }
        });

// Start the processing thread; the main thread remains free (non-blocking)
streamProcessor.start();

> streamProcessor.join(); // Wait for the thread to finish (blocking)

Reviewer: Make consistent with other comments

```suggestion
// Wait for the thread to finish (blocking)
streamProcessor.join(); 
```
